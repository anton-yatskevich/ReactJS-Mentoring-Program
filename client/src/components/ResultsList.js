import React from 'react';
import PropTypes from 'prop-types';
import ResultsItem from './ResultsItem';

const ResultsList = ({ movies, onSelectMovieHandler }) => (
    <div className="results-item__container">
        {
            movies.map(item => (
                <ResultsItem {...item} key={item.id} onSelectHandler={onSelectMovieHandler} />))
        }
    </div>
);

ResultsList.propTypes = {
    movies: PropTypes.arrayOf(PropTypes.objectOf(
        PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number,
            PropTypes.arrayOf(PropTypes.string)
        ])
    )),
    onSelectMovieHandler: PropTypes.func
};

ResultsList.defaultProps = {
    movies: [],
    onSelectMovieHandler: null
};

export default ResultsList;
