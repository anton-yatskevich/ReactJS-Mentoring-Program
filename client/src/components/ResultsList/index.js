import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import noop from 'lodash.noop';
import ResultsItem from '../ResultsItem';
import MovieSchema from '../../constants/MovieSchema';

const ResultsList = ({ movies, selectMovie }) => (
    <Container>
        {
            movies.map(item => (
                <ResultsItem movie={item} key={item.id} onSelectHandler={selectMovie} />))
        }
    </Container>
);

ResultsList.propTypes = {
    movies: PropTypes.arrayOf(MovieSchema),
    selectMovie: PropTypes.func
};

ResultsList.defaultProps = {
    movies: [],
    selectMovie: noop
};

const Container = styled.div`
    display: flex;
    flex-wrap: wrap;
    max-width: 1200px;
    margin: 0 auto;
    justify-content: space-around;
    padding: 40px 0;
`;

export default ResultsList;
