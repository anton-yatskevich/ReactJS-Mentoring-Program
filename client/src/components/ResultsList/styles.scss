.results__container {
    display: flex;
    flex-wrap: wrap;
    max-width: 1200px;
    margin: 0 auto;
    justify-content: space-around;
    padding: 40px 0;
}