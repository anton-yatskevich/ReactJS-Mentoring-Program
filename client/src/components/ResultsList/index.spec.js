import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ResultsList from './index';

describe('ResultsList component', () => {
    it('should be rendered with default props', () => {
        const ResultsListComponent = shallow(<ResultsList />);
        expect(toJson(ResultsListComponent)).toMatchSnapshot();
    });

    it('should be rendered with passed results', () => {
        const props = {
            movies: [{ id: 1 }, { id: 2 }]
        };
        const ResultsListComponent = shallow(<ResultsList {...props} />);
        expect(ResultsListComponent.find('ResultsItem').length).toEqual(2);
    });
});
