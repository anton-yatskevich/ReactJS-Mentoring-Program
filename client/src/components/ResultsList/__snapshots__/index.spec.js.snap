// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsList component should be rendered with default props 1`] = `
<div
  className="results__container"
/>
`;
